app/src/main/java/com/llamalabb/TTT3D/tictactoe/TicTacToeActivity.kt

```diff
@@ -12,8 +12,25 @@ import kotlinx.android.synthetic.main.cell_layout.view.*
 class TicTacToeActivity : AppCompatActivity(), TicTacToeContract.View {
     override var presenter: TicTacToeContract.Presenter = TicTacToePresenter(this)
 
+    private val boardListener = object : TicTacToeContract.BoardListener {
+        override fun cellClickListener(index: Int) {
+            presenter.handleClickOnIndex(index)
```

Comment:
The adapter is still receiving an implicit reference to the presenter here. So the presenter is in essence "inside of the adapter" still.